drainer/syncer.go

```diff
@@ -323,8 +323,15 @@ func (s *Syncer) sync(executor executor.Executor, jobChan chan *job, executorIdx
 		sqls = sqls[0:0]
 		args = args[0:0]
 		commitTSs = commitTSs[0:0]
-		s.lastSyncTime = time.Now()
-		lastSyncTime = time.Now()
+		now := time.Now()
+		s.lastSyncTime = now
+		if !maxExecutionWaitTimer.Stop() {
+			if !draineTimer {
+				<-maxExecutionWaitTimer.C
```

Comment:
```go
select {
  case <-maxExecutionWaitTimer.C:
  default:
  }
```

Can we use `select` here to avoid managing the extra boolean variable?

Reply:
I don't think we can assume we can trigger the `case <-maxExecutionWaitTimer.C:` absolutely, that means then the event may be pushed into the `chan` by another goroutine.
better make it explicitly when we need to drain it as the docs.

Reply:
Do we need to set `draineTimer` back to `false` in other cases?

Reply:
https://github.com/pingcap/tidb-binlog/pull/863/commits/eada8b703002893b86e77a62769f9462e55d83ac